04-onion-routing.md

```diff
@@ -1407,6 +1438,117 @@ The _origin node_:
+classification is related to the node's current routing activity. Nodes MAY
+choose a window per their preferences - shorter windows being more reactive to 
+recent routing patterns, and longer windows aggregating trends over time.
+
+- `routing_window`: the period of time in which total routing revenue is 
+  assessed (default: 10 minutes * the maximum number of blocks a HTLC may be
+  held before the node will send `expiry_too_far`, as outlined in [Failure Messages](#failure-messages)).
+- `routing_revenue`: the sum of all fees paid to the node to forward HTLCs
+  over the interval [ `now` - `routing_window` ; `now` ].
+
+The total `effective_fees` that an individual peer has generated are assessed
+over a longer period of time to relate its reputation classification to the 
+routing activity of the node (represented by `routing_revenue`).
+
+- `reputation_window`: the period of time over which the node's `effective_fees`
+  are calculated to classify its reputation (recommended default = 10 * `routing_window`).
+- `reputation_revenue`: the total `effective_fees` of HTLCs forwarded by the peer
+  over the interval [ `now` - `reputation_window` ; `now` ] less the sum of 
+  `outstanding_risk` for all incoming, in-flight HLTCs from the peer.
+
+A peer is classified as having "good" local reputation iff `reputation_revenue` >= `routing_revenue`, 
+otherwise the peer's reputation is classified as "neutral".
+
+### Rationale
+
+A jamming attack can only be achieved with good reputation, as the resources 
+available to neutral reputation nodes are limited by resource bucketing. If a 
+node is attacked, it is guaranteed to have been paid at least its 
+`routing_revenue` over the previous `reputation_window`.
+
+Reputation is only negatively affected if a HTLC was endorsed to help protect 
+the network's ability to forward "unknown" HTLCs - those from low-activity or 
+new entrants that may very well be honest, but have not build up good 
+reputation through a history of consistent, good behavior.
+
+Intuitively if a HTLC is endorsed by a peer they have signaled that they expect 
+the HTLC to resolve honestly, they should be held accountable for that signal.
+By contrast, if a HTLC was not endorsed by the upstream peer, it should not 
+have a negative impact on reputation. In the case where one of these 
+"unknown" HTLCs succeeds within the reasonable resolution threshold, the 
+peer is still credited with fees because the HTLC resolved desirably. This 
+allows new, unendorsed entrants to slowly build reputation over time. The fee 
+contribution of unendorsed HTLCs is discounted by 50% to incentivise nodes to 
+endorse HTLCs.
+
+In flight HLTCs are included in reputation scoring to account for sudden changes
+in a peer's behavior. Even when good reputation is obtained, each HTLC choosing 
+to take advantage of that good reputation is treated as if it will be used to 
+inflict maximum damage. The expiry height of each incoming in flight HLTC is 
```

Comment:
Note the comment on the the necessity to integrate fee-bumping reserve, here the size of the worst-case second HTLC transaction (and its channel type) will have to be integrate in the “inflict maximum damage” than an adversary can inflict to a routing hop.

Reply:
This unnecessarily complicates the calculation, the reduction in reputation is severe enough

Reply:
You don’t choose the “calculation” of maximum damage that can be inflicted,  the adversary is doing it :)  Though yeah good to start with simple calculation.